src/ImageRunModal.jsx

```diff
@@ -32,135 +38,177 @@ const units = {
-                <TextInput aria-label={_("Container path")}
-                           placeholder={_("Container path")}
-                           value={item.containerPath || ''}
-                           onChange={value => onChange(idx, 'containerPath', value)} />
+                </FormGroup>
+            </Grid>
+        );
+    } else {
+        return (
+            <EmptyState>
+                <EmptyStateBody>
+                    {_("No environment variables specified")}
+                </EmptyStateBody>
+            </EmptyState>
+        );
+    }
+};
 
-                <Button variant='secondary'
-                        className={"btn-close" + (idx === 0 && !item.containerPath && !item.hostPath ? ' invisible' : '')}
-                        aria-label={_("Remove item")}
-                        isSmall
-                        icon={<CloseIcon />}
-                        onClick={() => removeitem(idx)} />
-                <Button variant='secondary'
-                        className="btn-add"
-                        onClick={additem}
-                        isSmall
-                        icon={<PlusIcon />}
-                        aria-label={_("Add item")} />
-            </InputGroup>
-            <InputGroup className='ct-input-group-spacer-sm'>
-                <FormSelect className='pf-c-form-control'
-                            aria-label={_("Mode")}
-                            value={item.mode}
-                            onChange={value => onChange(idx, 'mode', value)}>
-                    <FormSelectOption value='ro' key='ro' label={_("ReadOnly")} />
-                    <FormSelectOption value='rw' key='rw' label={_("ReadWrite")} />
-                </FormSelect>
+const Volume = ({ id, item, onChange, idx, removeitem, additem, options, itemCount }) => {
+    if (itemCount) {
+        return (
+            <Grid hasGutter id={id}>
+                <FormGroup className="pf-m-3-col-on-sm" label={_("Host path")} fieldId={id + "-host-path"}>
+                    <FileAutoComplete id={id + "-host-path"}
+                        value={item.hostPath || ''}
+                        onChange={ value => onChange(idx, 'hostPath', value) } />
+                </FormGroup>
+                <FormGroup className="pf-m-3-col-on-sm" label={_("Container path")} fieldId={id + "-container-path"}>
+                    <TextInput id={id + "-container-path"}
+                        value={item.containerPath || ''}
+                        onChange={value => onChange(idx, 'containerPath', value)} />
+                </FormGroup>
+                <FormGroup className="pf-m-2-col-on-sm" label={_("Mode")} fieldId={id + "-mode"}>
+                    <Checkbox id={id + "-mode"}
+                        label={_("Writable")}
+                        value={item.mode == "rw"}
+                        onChange={value => onChange(idx, 'mode', value ? "rw" : "ro")} />
+                </FormGroup>
```

Comment:
Toggling the button, didn't work for me, nor is it checked by default.

Reply:
Fixed.